manuskript/exporter/pandoc/__init__.py

```diff
@@ -75,7 +75,10 @@ def convert(self, src, args, outputfile=None):
                 args.append("--variable={}:{}".format(var, item.text().strip()))
 
         # Add title metatadata required for pandoc >= 2.x
-        args.append("--metadata=title:{}".format(mainWindow().mdlFlatData.item(0, 0).text().strip()))
+        title = " "   # blank prevents args.append() crash if no project title
+        if mainWindow().mdlFlatData.item(0, 0):
+            title = mainWindow().mdlFlatData.item(0, 0)
+        args.append("--metadata=title:{}".format(title.text().strip()))
```

Comment:
`title.text().strip()` would crash if there is no title set, since `title = " "` makes it a string, and python string doesn't have the `.text()` method.
Also, having the title be a space is useless if you call strip() afterwards on it.
I'd suggest this maybe : 
```
       title = "-"   # blank prevents args.append() crash if no project title
        if mainWindow().mdlFlatData.item(0, 0):
            title = mainWindow().mdlFlatData.item(0, 0).text().strip()
        args.append("--metadata=title:{}".format(title))
```

Reply:
Thanks for reviewing my patch.  And your comments and understanding of the code are spot on.

My goal was to set the title to an empty string in the case where a user did not provide a title for the Manuskript project.  Would it help if I added this goal to the comments?

Or do you think it is better to set the title to a minus sign "-" in the case where the user does not provide a title?

Reply:
If an empty string works, then go for it. I maybe misunderstood the comment "blank prevents args.append crash if no project title". I thought it meant that you put a blank (a space) in the title because otherwise (if it's an empty string) then it would crash, potentially because "--metadata=title:" as argument would take the next argument in the command line as the title, thus breaking the command. But if it actually works with an empty string, then you can use that.

Actually, if the title isn't set, you could maybe not add the metadata argument at all, I don't think pandoc requires it for it to work. So... 
```
        # Only set the title metadata if a title has been set for the project.
        if mainWindow().mdlFlatData.item(0, 0):
            title = mainWindow().mdlFlatData.item(0, 0).text().strip()
            args.append("--metadata=title:{}".format(title))
```

would that work ?

Reply:
Your propsed solution immediately above should work for pandoc v1.x.  The problem arises with pandoc v2.x as it requires either a "title" or "pagetitle" to be set in the metadata.  See near the bottom of this [issue 304 comment](https://github.com/olivierkes/manuskript/issues/304#issuecomment-370505755) and also PR #418.

Interestingly the warning in pandoc v2.x indicates the "title" must be specified and non-empty.  However pandoc v2.x works as long as "title" is specified even if it is non-empty.  This makes me wonder if your minus sign "-" solution might be better as it may work better with future pandoc versions.  What do you think?

Reply:
Or I could set the empty title to "Untitled" as this seems to be what is suggested by the pandoc v2.x warning.

Reply:
Yes, I was just writing to suggest "Untitled" then saw pandoc suggests the same then you suggested it as well, so I guess we have consensus :)

Reply:
Thanks for your input.  I will update the MR accordingly.